### ISSUE TITLE
Create unique user sessions

### ISSUE THREAD

**Issue by nguyenmq** [nguyenmq/yt_box#21]
When a user logs in, that user's session should be unique for its lifetime. This is mostly to prevent users from aliasing each other and doing malicious things to one another.

**Comment by nguyenmq** [comment]
@crawfordia @Phillter 

tl;dr
There's a new user_id branch and I need you guys to test it.

I pushed a "user_ids" branch containing a fairly large overhaul of how users are handled throughout the system. Every unique user name gets an id now and each of those name and id combos is stored inside a sqlite database that's accessed and maintained by yt_player. This means everything references a user based on his/her integer id now. I need you guys to give this branch a try and make sure I didn't break anything.

When a user logs in, that name is placed inside a set inside yt_player. The next person who tries to login with the same name will get rejected because the name is already in use. Now this does create a couple of small problems that I'm not sure needs to be addressed right now. First, if the backend crashes (which you'll more likely see during development), any logged in users will not be marked as being "logged in" the next time you start backend. This means two user can share the same user name. The second problem is if backend were to ever become a long-running process, a user who never logs out will have a lock on his user name. Furthermore, once his cookie expires and he is forced to log back in, he won't be able to use his name again because we think it's still in use.

A possible solution to the first problem is to force everyone to log back in if the backend were to unexpectedly die. It seems reasonable during production, but that can be annoying during development. For the second problem, we would need to keep track of last-access timestamps in the database. If a user hasn't be active for some amount of time and someone tries to log in with that name, we'll consider the first person logged out and give the name to the new user.

**Comment by crawfordia** [comment]
By manually going back to the login page you can claim multiple user names.

**Comment by nguyenmq** [comment]
Nice catch. Will work on a fix for it soon.

**Comment by nguyenmq** [comment]
Pushed some changes to implement what we talked about today. Before trying this out, you'll need to delete the existing `config/yt_box.db` file if it's there so that the database can be recreated with the tables and columns it expects.

User names still need to be unique though. I think there's some value there in case we want to manually allow reclaiming of ids and user names in the future from an admin account. The last access date/time stamp is also still there, which allows for purging of inactives if we ever _really_ want it, although that would make it a little tougher to join user_ids to other tables. The session cookie also lasts for a year now.

One thing we didn't talk about today is there's a submissions table that records all the videos submitted. A future feature I want to implement is the ability to browse and search past submissions. This sets the groundwork for that. For fun, you can open the sqlite database and match users to submissions by running:

``` sql
select * from users join submissions using (user_id);
```

I encourage you to brush up on your sql knowledge. I super promise it'll be fun. All the sql stuff is in yt_player.py where you can look at all the sql queries we're currently using. I'm using the convention that all sql key words in a query are in all caps.
